snp_measurement/src/vmsa.rs

```diff
@@ -0,0 +1,49 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use log::trace;
+use oak_sev_guest::vmsa::{calculate_rdx_from_fms, Vmsa, VmsaPage};
+use x86_64::PhysAddr;
+
+/// The CPU family of the vCPU we expect to be running on.
+const CPU_FAMILY: u8 = 6;
+
+/// The CPU model of the vCPU we expect to be running on.
+const CPU_MODEL: u8 = 0;
+
+/// The stepping of the vCPU we expect to be running on.
+const CPU_STEPPING: u8 = 0;
+
+/// The guest-physical address of the VMSA page.
+///
+/// The current implementation uses the same fixed address for all VMSA pages. It is calculated as
+/// the start-address of the last 4KiB page that can be addressed within the allowed physical bits.
+///
+/// For AMD "Milan" CPUs the maximum supported physical memory bit-width is 48 when SEV-SNP is
+/// enabled.
+#[allow(unused)]
+pub const VMSA_ADDRESS: PhysAddr = PhysAddr::new(0xFFFF_FFFF_F000);
```

Comment:
I wonder if `(1 << 48) - 0x1000` would be worth using here, as it would make the address more explicit at the cost of slightly lower readability.

Reply:
Good idea. I ended up goind with `(1 << 48) - Size4KiB::SIZE` to try and make it as readable as possible.